corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -61,10 +61,13 @@ function hqDefine(path, dependencies, moduleAccessor) {
                 'underscore': '_',
                 'clipboard/dist/clipboard': 'Clipboard',
                 'ace-builds/src-min-noconflict/ace': 'ace',
+                'chai/chai': 'chai',
                 'DOMPurify/dist/purify.min': 'DOMPurify',
+                'mocha/mocha': 'mocha',
                 'moment/moment': 'moment',
                 'crypto-js/crypto-js': 'CryptoJS',
                 'hqwebapp/js/lib/modernizr': 'Modernizr',
+                'sinon/pkg/sinon': 'sinon',
```

Comment:
Is this adding test modules to production code? If yes, would it be possible to move them to a place where they are only included when running tests? For example, would it be possible to move them into [corehq/apps/mocha](https://github.com/dimagi/commcare-hq/tree/870433df22141d411483d9639eb2babe561a4f89/corehq/apps/mocha)?

Reply:
It isn't causing the test modules to get including in prod code, it's just configuration. It's configuration that's only applicable to tests, so I can separate it into another object if that'd be clearer, but it can't be moved into a different file.

Reply:
No need to change if it's only applicable to tests.

I think I understand what this is doing now. What would happen if a production module used `hqDefine` and required a dependency like (for example) `sinon`? Would that cause `sinon` to be `undefined` within the `hqDefine` scope (when running in production) since `sinon` is a test dependency and is not available in production, or would it cause `sinon` to be loaded in production?

Reply:
The second outcome. All modules are always available for import.

Reply:
Turns out this is not quite true. Locally, all modules are always available. However, if a production module depends on a non-production module, the R.js optimization will fail, which will block deploy - see https://github.com/dimagi/commcare-hq/pull/32459.